core/src/test/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoEntityExpressionValidatorSpec.scala

```diff
@@ -0,0 +1,193 @@
+package org.broadinstitute.dsde.rawls.entities.datarepo
+
+import bio.terra.datarepo.model.{ColumnModel, TableModel}
+import cromwell.client.model.ValueType.TypeNameEnum
+import cromwell.client.model.{ToolInputParameter, ValueType}
+import org.broadinstitute.dsde.rawls.RawlsTestUtils
+import org.broadinstitute.dsde.rawls.dataaccess.slick.TestDriverComponent
+import org.broadinstitute.dsde.rawls.entities.base.ExpressionValidator
+import org.broadinstitute.dsde.rawls.expressions.DataRepoExpressionFixture
+import org.broadinstitute.dsde.rawls.jobexec.MethodConfigResolver.{GatherInputsResult, MethodInput}
+import org.broadinstitute.dsde.rawls.model.{AgoraMethod, Attributable, AttributeString, MethodConfiguration}
+import org.scalatest.FlatSpec
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.time.{Seconds, Span}
+
+import scala.collection.JavaConverters._
+
+class DataRepoEntityExpressionValidatorSpec extends FlatSpec with TestDriverComponent with RawlsTestUtils with ScalaFutures with DataRepoEntityProviderSpecSupport with DataRepoExpressionFixture {
+
+  def toExpressionMap(expressions: Seq[String]): Map[String, AttributeString] =
+    expressions.map { expr => expr.toString -> AttributeString(expr) }.toMap
+
+  def toMethodInput(tuple: (String, AttributeString)): MethodInput = tuple match {
+    case (name, expr) => {
+      val valueType = new ValueType()
+      valueType.setTypeName(TypeNameEnum.STRING)
+      val input = new ToolInputParameter()
+      input.setName(name)
+      input.setValueType(valueType)
+      MethodInput(input, expr.value)
+    }
+  }
+
+  def toMethodInputs(methodConfiguration: MethodConfiguration) : Seq[MethodInput] = {
+    methodConfiguration.inputs.map(toMethodInput).toSeq
+  }
+
+  def toGatherInputs(exprs: Map[String, AttributeString]): GatherInputsResult = {
+    val methodInputs = exprs.map(toMethodInput)
+    GatherInputsResult(methodInputs.toSet, Set(), Set(), Set())
+  }
+
+  val rootTableName = "rootTable"
+  // todo: how to keep this in sync with the fixtures...
```

Comment:
Does anyone have thoughts on this? These columns need to match up with the input expressions defined in the fixture in order for the tests to validate the input expressions correctly. Other than just leaving them here and hoping that they stay in sync, my other idea was adding some of these vals to `DataRepoExpressionFixture` (`rootTableName`, `rootTableColumns`, and `tables`) so that they're at least all defined in the same place. We could take it one step further and try to construct the lists of valid and invalid expressions using the values in `rootTableColumns` but that'd complicate things a bit

Reply:
I would just move things to the fixture to keep them in the same place